From: Markus Koschany <apo@debian.org>
Date: Tue, 6 Dec 2016 18:04:02 +0100
Subject: gold linker

This is a bug in binutils which will probably not fixed in time for Stretch.
Work around the build failure by switching to Debian's default linker instead
of the gold linker. This patch can be reverted as soon as #847356 is fixed.

Debian-Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846921
Forwarded: not-needed
---
 rts/build/cmake/TestCXXFlags.cmake | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/rts/build/cmake/TestCXXFlags.cmake b/rts/build/cmake/TestCXXFlags.cmake
index 7b338d3..d566c55 100644
--- a/rts/build/cmake/TestCXXFlags.cmake
+++ b/rts/build/cmake/TestCXXFlags.cmake
@@ -162,21 +162,21 @@ if   (CMAKE_COMPILER_IS_GNUCXX)
 	)
 	set(hasGold FALSE)
 	if (${linkerVersion} MATCHES "gold")
-		set(hasGold TRUE)
+        set(hasGold FALSE)
 	endif()
 
 
-	if    (NOT hasGold AND NOT WIN32) #FIND_PROGRAM fails in crosscompile environments (it detects the native ld.gold)
+    #if    (NOT hasGold AND NOT WIN32) #FIND_PROGRAM fails in crosscompile environments (it detects the native ld.gold)
 		# since gcc 4.8 it is possible to switch the linker via that argument
-		CHECK_CXX_ACCEPTS_FLAG("-fuse-ld=gold" HAS_USE_LD)
-		IF    (HAS_USE_LD)
-			FIND_PROGRAM(LD_GOLD ld.gold)
-			if    (LD_GOLD)
-				set(hasGold TRUE)
-				set(LDGOLD_CXX_FLAGS "-fuse-ld=gold")
-			endif ()
-		EndIf ()
-	endif ()
+        #CHECK_CXX_ACCEPTS_FLAG("-fuse-ld=gold" HAS_USE_LD)
+        #IF    (HAS_USE_LD)
+        #		FIND_PROGRAM(LD_GOLD ld.gold)
+        #	if    (LD_GOLD)
+        #		set(hasGold TRUE)
+        #		set(LDGOLD_CXX_FLAGS "-fuse-ld=gold")
+        #	endif ()
+        #	EndIf ()
+        #endif ()
 
 	if    (hasGold)
 		set(LDGOLD_FOUND TRUE)
